src/pocketmine/block/SnowLayer.php

```diff
@@ -50,37 +50,48 @@ public function getToolType(){
 		return Tool::TYPE_SHOVEL;
 	}
 
-
 	public function place(Item $item, Block $block, Block $target, $face, $fx, $fy, $fz, Player $player = null){
 		$down = $this->getSide(0);
 		if($down->isSolid()){
-			$this->getLevel()->setBlock($block, $this, true);
-
-			return true;
+			if($down->getId() === $this->getId() && $down->getDamage() <= 7){
+				$down->setDamage($down->getDamage() + 1);
+				// $this->getLevel()->setBlock($down, $down, true);
+				
+				return true;
+			}
+			else{
```

Comment:
Ou, i see them now.. Wait, i'll create a new pr after you maybe merge the new pr for the cacti + sugarcane, so it will stay clean